cmd/machine-config-daemon/start.go

```diff
@@ -131,6 +133,9 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 	stopCh := make(chan struct{})
 	defer close(stopCh)
 
+	// Start local metrics listener
+	go daemon.StartMetricsListener(startOpts.metricsURL)
```

Comment:
This should be taking the stop-channel (or even better, a context) for graceful termination.